src/app/server/Mdns.cpp

```diff
@@ -55,30 +55,24 @@ bool HaveOperationalCredentials()
     ChipLogProgress(Discovery, "Failed to find a valid admin pairing. Node ID unknown");
     return false;
 }
+} // namespace
 
-// Requires an 8-byte mac to accommodate thread.
-chip::ByteSpan FillMAC(uint8_t (&mac)[8])
+void OnPlatformEvent(const DeviceLayer::ChipDeviceEvent * event)
 {
-    memset(mac, 0, 8);
-#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    if (chip::DeviceLayer::ThreadStackMgr().GetPrimary802154MACAddress(mac) == CHIP_NO_ERROR)
-    {
-        ChipLogDetail(Discovery, "Using Thread extended MAC for hostname.");
-        return chip::ByteSpan(mac, 8);
-    }
-#endif
-    if (DeviceLayer::ConfigurationMgr().GetPrimaryWiFiMACAddress(mac) == CHIP_NO_ERROR)
+    if (event->Type == DeviceLayer::DeviceEventType::kMdnsPlatformInitialized)
     {
-        ChipLogDetail(Discovery, "Using wifi MAC for hostname");
-        return chip::ByteSpan(mac, 6);
+        if (event->MdnsPlatformInitialized.IsInitialized)
```

Comment:
Is there any use case for sending `kMdnsPlatformInitialized` event with `IsInitialized == false`? If not, I suggest removing the event argument as the event type carries all the relevant information.

Reply:
When implementing it I thought about adding some more information and functionality, but then I changed my mind. Now it is definitely not needed.